### PR TITLE
[None][feat] Support Eagle3 on Mistral Large3

### PR DESCRIPTION
**Pull request by byshiue**
## Description

This PR continues the work of PR **NVIDIA/TensorRT-LLM#9820**. This PR adds the Eagle3 supporting on Mistral Large3.

## Summary by CodeRabbit

* **New Features**
  * Added Eagle3 speculative decoding support for Mistral Large 3 model.
  * Added `--eagle3_model_arch` command-line argument to select architecture (llama3 or mistral_large3).

* **Bug Fixes**
  * Improved draft model loading for multimodal models.
  * Enhanced attention metadata handling in speculative decoding loops.
  * Fix accuracy issue of Mistral Large3 FP8

* **Documentation**
  * Added advanced LLM configuration examples for Mistral Large 3 with Eagle3 decoding.

### CODE REVIEW

**Review by syuoni — APPROVED** [review]

File: tensorrt_llm/_torch/models/checkpoints/mistral/config_loader.py [hunk 1]
```diff
@@ -301,7 +304,31 @@ def load(self, checkpoint_dir: str, **kwargs) -> ModelConfig:
...
+                    "model.layers.*.self_attn.q_a_proj*",
+                    "model.layers.*.self_attn.kv_a_proj_with_mqa*",
+                ]
+        elif hf_quant_config.get("quant_method") == "fp8":
+            # Used for Eagle3 weight
+            if hf_quant_config.get("weight_block_size", []):
+                quant_config.quant_algo = QuantAlgo.FP8_BLOCK_SCALES
+                quant_config.exclude_modules = ["*kv_b_proj*", "*k_b_proj*", "*eh_proj"]
+
+                block_size = hf_quant_config.get("weight_block_size", [])
```
Comment (hypdeb) [inline tensorrt_llm/_torch/models/checkpoints/mistral/config_loader.py]: The default value of `[]` does not seem required here either. The `assert` on the next line will stop the program here no matter if it's `[]` or `None`.

Comment (hypdeb) [reply tensorrt_llm/_torch/models/checkpoints/mistral/config_loader.py]: But you might need to change the assert to `assert block_size is not None and tuple(block_size) == (128, 128)`. It's a bit more verbose but I think more explicit too.

File: tensorrt_llm/_torch/models/modeling_multimodal_utils.py [hunk 2]
```diff
@@ -339,7 +339,7 @@ def fuse_input_embeds(
             "cause token count mismatches within the inference batch.")
 
     text_embed = embedding_layer(input_ids[text_token_indices])
-    input_embeds = torch.empty(input_ids.shape[0],
+    input_embeds = torch.zeros(input_ids.shape[0],
```
Comment (hypdeb) [inline tensorrt_llm/_torch/models/modeling_multimodal_utils.py]: Is there an uninitialized read issue that prompts this change?

File: tensorrt_llm/_torch/models/modeling_speculative.py [hunk 3]
```diff
@@ -380,6 +382,153 @@ def apply_eagle3_fc(self, hidden_states: torch.Tensor) -> torch.Tensor:
         return hidden_states
 
 
+class MistralLarge3DraftModel(DecoderModel):
+
+    def __init__(
+        self,
+        model_config,
```
[the remark this answers is not in the document]
Comment (byshiue) [reply tensorrt_llm/_torch/models/modeling_speculative.py]: Updated. Thank you to remind.

File: tensorrt_llm/_torch/models/checkpoints/mistral/config_loader.py [hunk 4]
```diff
@@ -301,7 +304,31 @@ def load(self, checkpoint_dir: str, **kwargs) -> ModelConfig:
...
-                quant_config.exclude_modules = ["*q_a_proj*", "*kv_a_proj_with_mqa*"]
+                quant_config.exclude_modules = [
+                    "language_model.model.layers.*.self_attn.q_a_proj*",
+                    "language_model.model.layers.*.self_attn.kv_a_proj_with_mqa*",
+                    "model.layers.*.self_attn.q_a_proj*",
+                    "model.layers.*.self_attn.kv_a_proj_with_mqa*",
+                ]
+        elif hf_quant_config.get("quant_method") == "fp8":
+            # Used for Eagle3 weight
+            if hf_quant_config.get("weight_block_size", []):
```
[the remark this answers is not in the document]
Comment (byshiue) [reply tensorrt_llm/_torch/models/checkpoints/mistral/config_loader.py]: Updated. Thank you to remind.

File: tensorrt_llm/_torch/models/checkpoints/mistral/config_loader.py [hunk 5]
```diff
@@ -301,7 +304,31 @@ def load(self, checkpoint_dir: str, **kwargs) -> ModelConfig:
...
                 quant_config.quant_algo = QuantAlgo.FP8_BLOCK_SCALES
                 quant_config.group_size = 128
-                quant_config.exclude_modules = ["*q_a_proj*", "*kv_a_proj_with_mqa*"]
+                quant_config.exclude_modules = [
+                    "language_model.model.layers.*.self_attn.q_a_proj*",
+                    "language_model.model.layers.*.self_attn.kv_a_proj_with_mqa*",
+                    "model.layers.*.self_attn.q_a_proj*",
+                    "model.layers.*.self_attn.kv_a_proj_with_mqa*",
+                ]
+        elif hf_quant_config.get("quant_method") == "fp8":
```
Comment (lfr-0531) [inline tensorrt_llm/_torch/models/checkpoints/mistral/config_loader.py]: Looks like these two parts have the same settings. Can they be combined into one?
```
if hf_quant_config.get("weight_block_size") or hf_quant_config.get("activation_scheme", None) == "static":
    quant_config.quant_algo = QuantAlgo.FP8_BLOCK_SCALES
    quant_config.exclude_modules = ["*kv_b_proj*", "*k_b_proj*", "*eh_proj"]
    block_size = hf_quant_config.get("weight_block_size", (128, 128))
    assert tuple(block_size) == (128, 128), (
            f"FP8_BLOCK_SCALES only supports block_size=(128,128), current block_size: {block_size}"
                )
    quant_config.group_size = block_size[0]
```

File: tensorrt_llm/_torch/pyexecutor/model_loader.py [hunk 6]
```diff
@@ -277,18 +277,22 @@ def init_meta_tensor(t: torch.Tensor):
 
                 if self.spec_config is not None and self.spec_config.spec_dec_mode.need_load_draft_weights(
                 ):
+                    if getattr(model, "is_multimodal", False):
```
[the remark this answers is not in the document]
Comment (chang-l) [reply tensorrt_llm/_torch/pyexecutor/model_loader.py]: Instead of changing the general model loader logic for VLM models, can we handle the re‑direction within the VLM model file itself? For example: https://github.com/NVIDIA/TensorRT-LLM/blob/e6187d8109bc7107d6e0419d44e674e218e48df0/tensorrt_llm/_torch/models/modeling_mistral.py#L421-L425 or something like: 
```
@property
def draft_model(self):
    ...
```

**Review by QiJune — APPROVED** [review]
LGTM

**Review by lfr-0531 — APPROVED** [review]

**Review by jdebache — APPROVED** [review]

**Comment by byshiue** [comment]
/bot skip --comment "the only failed test test_auto_scaling.py::test_service_discovery[etcd-load_balancing] is wavied in main branch in PR 10161"
